e2e/api/steps.js

```diff
@@ -3,91 +3,221 @@ const assert = require('assert').strict;
+      header: 'You\'ve decided to proceed with the claim',
+      body: 'We\'ll review the case. We\'ll contact you to tell you what to do next.'
+    });
   }
 };
 
-const assertValidData = async (eventName, pageId, eventData, expectedDataSetByCallback = {}) => {
-  caseData = Object.assign(caseData, eventData);
+const assertValidData = async (pageId) => {
+  const validDataForPage = data[eventName].valid[pageId];
+  caseData = {...caseData, ...validDataForPage};
+
   const response = await request.validatePage(eventName, pageId, caseData);
   const responseBody = await response.json();
-  caseData = Object.assign(caseData, expectedDataSetByCallback);
+
+  if (response.status !== 200) {
+    console.log(responseBody);
+  }
 
   assert.equal(response.status, 200);
   assert.deepEqual(responseBody.data, caseData);
 };
 
-const assertCallbackError = async (eventName, pageId, eventData, expectedErrorMessage) => {
-  caseData = Object.assign(caseData, eventData);
-  const response = await request.validatePage(eventName, pageId, caseData);
+const assertCallbackError = async (pageId, eventData, expectedErrorMessage) => {
+  const response = await request.validatePage(eventName, pageId, {...caseData, ...eventData});
   const responseBody = await response.json();
 
   assert.equal(response.status, 422);
   assert.equal(responseBody.message, 'Unable to proceed because there are one or more callback Errors or Warnings');
   assert.equal(responseBody.callbackErrors[0], expectedErrorMessage);
 };
 
-const assertSubmittedEvent = async (eventName, expectedState, submittedCallbackResponse) => {
+const assertSubmittedEvent = async (expectedState, submittedCallbackResponseContains) => {
   const response = await request.submitEvent(eventName, caseData, caseId);
   const responseBody = await response.json();
 
+  if (response.status !== 201) {
+    console.log(responseBody);
+  }
+
   assert.equal(response.status, 201);
   assert.equal(responseBody.state, expectedState);
   assert.equal(responseBody.callback_response_status_code, 200);
-  assert.equal(responseBody.after_submit_callback_response.confirmation_header.includes(submittedCallbackResponse.header), true);
-  assert.equal(responseBody.after_submit_callback_response.confirmation_body.includes(submittedCallbackResponse.body), true);
+  assert.equal(responseBody.after_submit_callback_response.confirmation_header.includes(submittedCallbackResponseContains.header), true);
+  assert.equal(responseBody.after_submit_callback_response.confirmation_body.includes(submittedCallbackResponseContains.body), true);
 
-  caseData = Object.assign(caseData, responseBody.case_data);
+  caseData = {...caseData, ...responseBody.case_data};
   if (eventName === 'CREATE_CLAIM') {
     caseId = responseBody.id;
     console.log('Case created: ' + caseId);
   }
 };
+
+// Mid event will not return case fields that were already filled in another event if they're present on currently processed event.
```

Comment:
hmm this is the bug we keep talking about in our journies isn't it. Maybe one to raise with CCD!